.github/workflows/validate_mi_mutation.yml

```diff
@@ -0,0 +1,29 @@
+name: Validate mi driver mutations
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  validate:
+    name: Validate mi driver mutations
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v2
+
+    - name: Setup python3
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install build dependencies
+      run: python -m pip install -r python_build_requirements.txt
+
+    - name: Validate mi driver metadata mutations 
+      run: |
+        python source/codegen/format_mi_metadata.py source/codegen/metadata
+    
+    - name: Check for files dirtied by codegen
+      run: |
+        git diff --ignore-space-at-eol --exit-code
```

Comment:
Note: It was tricky getting the end of lines to match up completely between here and the hapigen output... especially when this runs on Windows vs. Ubuntu so we added `--ignore-space-at-eol` to just not care about the line endings.

Reply:
The `\r\n` code is definitely wrong on linux unless there's some double-negative there that I'm not understanding. If it's not breaking the linux build, I guess it's OK to leave it if you can't figure this out. But it's confusing enough that I bet someone will run into it again.

Reply:
Not necessary in this PR, consider adding a `.gitattributes` to address the line ending behavior. See [here.](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings).

Reply:
I removed the `--ignore-space-at-eol` and it should still work. I think I was getting confused by the newline stuff and at some point put this in for it to work. But it seems like and should work without it.